cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/prolog

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -e
+exec &>>"/var/log/parallelcluster/slurm_prolog_epilog.log"
+timestamp=`date +"%Y-%m-%d %H:%M:%S,%3N"`
+echo "$timestamp Adding nodes for job ${SLURM_JOB_ID} to /etc/hosts"
+myip=$(hostname -I)
+added=false   # If the nodes have been added to /etc/hosts
+no_add=false   # If it is not a safe line in /etc/hosts to add the nodes
+HOSTS=`echo "$SLURM_NODE_ALIASES" |awk 'BEGIN{RS=","; FS=":"}; {gsub(/\[|\]/,"",$2); print $2,$1,$3}'`
+# Read /etc/hosts line by line and output the lines to /etc/tmphosts. Then overwrite /etc/hosts with /etc/tmphosts
+test -f /etc/tmphosts && rm /etc/tmphosts
```

Comment:
if you need to use a temporary file let's create it under /tmp and let's make sure permissions and ownership are preserved.

Reply:
I will keep that in mind.
The new code does not need temporary file anymore